megaavr/cores/dxcore/wiring_analog.c

```diff
@@ -871,3 +873,19 @@ uint8_t digitalPinToTimerNow(uint8_t p) {
     }
   }
 */
+
+// MALS-98
+void analogReadSampleDelay(uint8_t delay) {
+  if (~(ADC0.COMMAND & ADC_STCONV_bm) && delay >= 0 && delay <= 15) {
+    ADCConfig.sampleDelay = delay;
+    ADC0.CTRLD |= delay;
```

Comment:
This won't quite be enough to set the delay. Suppose we have `ADC0.CTRLD = 0b11001101` and `delay = 6 = 0b0110`. Then just doing the `or`,
```c
ADC0.CTRLD |= delay; // After: ADC0.CTRLD = 0b11001111
```
Problem is that the 0 bits in delay that correspond with 1s in the current register value don't get zeroed. One solution would be to zero out all 4 sample delay bits and do the `or` operation:
```c
ADC0.CTRLD &= 0b11110000; // After: ADC0.CTRLD = 0b11000000
ADC0.CTRLD |= delay; // After: ADC0.CTRLD = 0b11000110
```